Refactor `useCombinedDomain` and add to lib package (#621)
silx-kit/h5web: src/stories/Home.stories.mdx

The library exposes a number of utility functions, which are documented below, as well as all the enums and types used by these utility functions.
If you use only the top-level visualization components, the only utilities you need are `getDomain` and/or `useDomain`.

#### getDomain

Find the min and max values contained in a array (or ndarray) of numbers. If `scaleType` is `ScaleType.Log` and the domain crosses zero, clamp the min to the first positive value or return `undefined` if the domain is not supported (i.e. `[-x, 0]`).

```ts
getDomain(values: ndarray | number[], scaleType: ScaleType = ScaleType.Linear): Domain | undefined

const linearDomain = getDomain([10, 5, -1]); // [-1, 10]
const logDomain = getDomain([-1, 2, 10], ScaleType.Log); // [2, 10]

const myArray = ndarray([0, 1, 2, 3], [2, 2]);
const myDomain = getDomain(myArray); // [0, 3]
```

#### getDomains

> Find the domains of multiple arrays (or ndarray) of numbers. Useful when dealing with auxiliary curves.

Reviewer: ```suggestion
Find the domains of multiple arrays (or ndarrays) of numbers. Useful when dealing with auxiliary curves.
```

```ts
getDomains(arrays: (ndarray | number[])[], scaleType: ScaleType = ScaleType.Linear): (Domain | undefined)[]

const linearDomains = getDomains([[-1, 5, 10], myArray]); // [[-1, 10], [0, 3]]
const logDomains = getDomains([[-1, 5, 10], myArray], ScaleType.Log); // [[2, 10], [0, 3]]
```

#### getCombinedDomain

Combine multiple domains into one. Useful to find the overall domain of a line visualization with auxiliary curves.

```ts
getCombinedDomain(domains: (Domain | undefined)[]): Domain | undefined

const combinedDomain = getCombinedDomain([[-1, 10], [0, 30]]]); // [-1, 30]
```

#### extendDomain

> Extend a domain by a given factor.

Reviewer: ```suggestion
Extend a domain by a given factor. The factor is scaled according to the given scale.
```

```ts

> extendDomain(bareDomain: Domain, extendFactor: number, isLog?: boolean): Domain

Reviewer: This declaration is not right.

const extendedDomain1 = extendDomain([0, 100], 0.5]); // [-50, 150]
const extendedDomain2 = extendDomain([10, 100], 1, ScaleType.Log); // approx. [1, 1000]
const extendedDomain3 = extendDomain([2, 2], 0.5); // [1, 3]
const extendedDomain4 = extendDomain([1, 1], 1, ScaleType.Log); // [0.1, 10]
```

#### computeVisSize

> Compute the optimal size for a visualization to fill the available space given an optional aspect ratio.

Reviewer: ```suggestion
Compute the optimal size for a visualization to fill the available space given a aspect ratio. If this aspect ratio is omitted, it will return the available size.
```

```ts
computeVisSize(availableSize: Size, aspectRatio?: number): Size | undefined

const size = computeVisSize({ width: 20, height: 10 }, 1.5); // { width: 15, height: 10 }
```

#### getDims

> Get the dimensions of a `ndarray` as a `{ rows, cols }` object.

Reviewer: Remind me why are we exporting this ? :joy:

Author: Hmm, I see no good reason 😂 - removing it now.
